### PR TITLE
add redux-form; searchbar, profile page, login

### PR DESCRIPTION
**Pull request by ChornayaDrakoshig**

### CODE REVIEW

File: client/src/components/pages/addpost-page/AddPostPage.jsx [hunk 1]
```diff
@@ -100,50 +98,45 @@ class AddPostPage extends React.Component {
     const defaulValueDescription = 'Input description of your post.';
 
     return (
```
Comment (akostiv) [inline client/src/components/pages/addpost-page/AddPostPage.jsx]: Refactor this component to use redux form as well?

File: client/src/components/pages/home-page/HomePage.jsx [hunk 2]
```diff
@@ -11,28 +11,39 @@ class HomePage extends React.Component {
...
-  renderPosts() {
-    if (this.props.posts)
-      return this.props.posts.map(
+  renderPostList() {
+    const searchQuery = this.props.searchQuery.trim().toLowerCase();
+
+    if (this.props.posts) {
+      let posts = [ ...this.props.posts];
+      if (searchQuery.length > 0) {
+        posts = posts.filter(post => (
```
Comment (akostiv) [inline client/src/components/pages/home-page/HomePage.jsx]: move filter logic into selector

File: client/src/components/pages/home-page/HomePage.jsx [hunk 3]
```diff
@@ -11,28 +11,39 @@ class HomePage extends React.Component {
...
+          post.Title.toLowerCase().indexOf(searchQuery) >= 0 ||
+          this.props.users[post.UserId].toLowerCase().indexOf(searchQuery) >= 0
```
Comment (akostiv) [inline client/src/components/pages/home-page/HomePage.jsx]: use regex to find matches

File: client/src/components/pages/home-page/index.jsx [hunk 4]
```diff
@@ -1,9 +1,14 @@
 import { connect } from 'react-redux';
+import { formValueSelector } from 'redux-form';
 import HomePage from 'components/pages/home-page/HomePage.jsx';
 
+const searchBarSelector = formValueSelector('searchBar');
+
 const mapStateToProps = (state) => ({
   user: state.user,
+  users: state.profile.usersById,
   posts: [ ...state.posts.posts].reverse(),
```
Comment (akostiv) [inline client/src/components/pages/home-page/index.jsx]: move this into selector

File: client/src/components/pages/myposts-page/MyPostsPage.jsx [hunk 5]
```diff
@@ -11,28 +11,35 @@ class MyPostsPage extends React.Component {
   }
 
   renderPostList = () => {
+    const searchQuery = this.props.searchQuery.trim().toLowerCase();
+
     if (this.props.posts) {
-      return this.props.posts.map(
+      let posts = [ ...this.props.posts];
+      if (searchQuery.length > 0) {
+        posts = posts.filter(post => (post.Title.toLowerCase().indexOf(searchQuery) >= 0))
```
Comment (akostiv) [inline client/src/components/pages/myposts-page/MyPostsPage.jsx]: move into selector

File: client/src/components/pages/profile-page/ProfilePage.jsx [hunk 6]
```diff
@@ -23,53 +24,50 @@ class ProfilePage extends React.Component {
...
-    if (user) {
-      this.state = {
-        newname: user.FirstName,
-        newsecondName: user.LastName,
-        newlogin: user.Login,
-      };
-    }
   }
 
   componentWillMount() {
```
Comment (akostiv) [inline client/src/components/pages/profile-page/ProfilePage.jsx]: make side effects in componentDidMount lifecycle hook

File: client/src/components/pages/profile-page/ProfilePage.jsx [hunk 7]
```diff
@@ -23,53 +24,50 @@ class ProfilePage extends React.Component {
...
-    const userId = +this.props.location.pathname.split(':')[1];
-    action({type : 'GET_USER_SAGA', payload: {userId}});
-    this.setState({
-      id: userId,
-    });
+    this.reloadProfile(this.props);
+  }
+  
+  componentWillReceiveProps(newProps) {
```
Comment (akostiv) [inline client/src/components/pages/profile-page/ProfilePage.jsx]: looks like a crutch, consider removing/reimplementing this logic

File: client/src/components/pages/profile-page/ProfilePage.jsx [hunk 8]
```diff
@@ -23,53 +24,50 @@ class ProfilePage extends React.Component {
...
   }
 
-  handleInputChange = (event) => {
-    this.setState({
-      [event.target.name]: event.target.value
-    });
+  reloadProfile = (props) => {
+    // TODO cnange to props.match.params.userId after removing ':' from url
+    const userId = +props.location.pathname.split(':')[1];
+    action({type : 'GET_USER_SAGA', payload: {userId}});
```
Comment (akostiv) [inline client/src/components/pages/profile-page/ProfilePage.jsx]: action should not contain 'SAGA' word

File: client/src/components/pages/profile-page/ProfilePage.jsx [hunk 9]
```diff
@@ -23,53 +24,50 @@ class ProfilePage extends React.Component {
...
+  handleSubmitChanges = (event) => {
+    event.preventDefault();
+
     this.setState({
       openDialog: true
     });
   }
 
-  change = () => {
+  change = (event) => {
```
Comment (akostiv) [inline client/src/components/pages/profile-page/ProfilePage.jsx]: move into https://redux-form.com/7.4.2/docs/api/reduxform.md/#-code-onsubmit-function-code-optional-?

File: client/src/components/pages/profile-page/confirmation-modal/ConfirmationForm.jsx [hunk 10]
```diff
@@ -0,0 +1,42 @@
+
+import * as React from 'react';
```
Comment (akostiv) [inline client/src/components/pages/profile-page/confirmation-modal/ConfirmationForm.jsx]: in some files react are imported in other way, consider using one way across whole app

File: client/src/assets/styles/theme/variables.less [hunk 11]
```diff
@@ -9,7 +9,7 @@
 /**Colors */
 @primary-color: @palette2;
 @secondary-color: @palette3;
-@inversed-color: @palette1;
+@color-for-dark-background: @palette1;
```
Comment (k-stasevich) [inline client/src/assets/styles/theme/variables.less]: no, not like this,
`@color-for-primary-background` or something `....-for-secondary-background`

File: client/src/components/common/header/login-modal/LoginForm.jsx [hunk 12]
```diff
@@ -0,0 +1,46 @@
+
```
Comment (k-stasevich) [inline client/src/components/common/header/login-modal/LoginForm.jsx]: excess linee

Comment (k-stasevich) [reply client/src/components/common/header/login-modal/LoginForm.jsx]: + login form isnt part of header, login form modal have to be layout independent thing

File: client/src/components/common/header/login-modal/LoginForm.jsx [hunk 13]
```diff
@@ -0,0 +1,46 @@
...
+    errors.password = 'Required'
+  } 
+  if (!values.password) {
+    errors.password = 'Required'
+  } 
+
+  return errors;
+}
+
+class LoginForm extends React.Component {
```
Comment (k-stasevich) [inline client/src/components/common/header/login-modal/LoginForm.jsx]: make it stateless component

File: client/src/components/containers/search-bar/SearchBar.jsx [hunk 14]
```diff
@@ -0,0 +1,21 @@
+import * as React from 'react';
+import { reduxForm, Field } from 'redux-form';
+import { getBEMClasses } from 'components/helpers/BEMHelper';
+import InputWithPlaceholder from 'components/common/input/InputWithPlaceholder.jsx';
+
+const searchBar = 'search-bar';
+const bemClasses = getBEMClasses([searchBar]);
+
+class SearchBar extends React.Component {
```
Comment (k-stasevich) [inline client/src/components/containers/search-bar/SearchBar.jsx]: stateless component

File: client/src/components/pages/profile-page/confirmation-modal/ConfirmationForm.jsx [hunk 15]
```diff
@@ -0,0 +1,42 @@
...
+const validate = values => {
+  const errors = {}
+  if (!values.password) {
+    errors.password = 'Required'
+  } 
+
+  return errors;
+}
+
+class ConfirmationForm extends React.Component {
```
Comment (k-stasevich) [inline client/src/components/pages/profile-page/confirmation-modal/ConfirmationForm.jsx]: stateless component

File: client/src/components/pages/profile-page/user-info-form/UserInfoForm.jsx [hunk 16]
```diff
@@ -0,0 +1,64 @@
+
```
Comment (k-stasevich) [inline client/src/components/pages/profile-page/user-info-form/UserInfoForm.jsx]: excess line

File: client/src/components/pages/profile-page/user-info-form/UserInfoForm.jsx [hunk 17]
```diff
@@ -0,0 +1,64 @@
...
+          />
+        </div>
+      </form>      
+    )
+  }
+}
+
+UserInfoForm = reduxForm({ form: 'userProfile', validate })(UserInfoForm);
+
+const mapStateToProps = state => {
```
Comment (k-stasevich) [inline client/src/components/pages/profile-page/user-info-form/UserInfoForm.jsx]: Containers have to be placed in separate file, do it for all cases that you affected in this PR

File: client/src/components/common/app/index.jsx [hunk 18]
```diff
@@ -1,11 +1,15 @@
...
 import Footer from 'components/common/footer/index.jsx';
+import Header from 'components/common/header/index.jsx';
 
 const App = (props) => {
   return (
     <React.Fragment>
       <div className="app-container">
-        {props.children}
+        <Header />
+          <div className="content">
```
Comment (k-stasevich) [inline client/src/components/common/app/index.jsx]: do code formatting, this line should be with same spaces as Header

File: client/src/components/containers/login-modal/LoginModal.jsx [hunk 19]
```diff
@@ -0,0 +1,45 @@
+import * as React from 'react';
```
Comment (k-stasevich) [inline client/src/components/containers/login-modal/LoginModal.jsx]: why we import react like this?

File: client/src/components/pages/myposts-page/MyPostsPage.jsx [hunk 20]
```diff
@@ -1,38 +1,40 @@
...
-      return this.props.posts.map(
-        post => (
-          <PostContainer
-            key={post.id}
-            post={post}
-          />
-        )
+    const { posts } = this.props;
+
+    if (posts) return posts.map(
```
Comment (k-stasevich) [inline client/src/components/pages/myposts-page/MyPostsPage.jsx]: ??? don't write if like this, especially when it is not one line of code

File: client/src/components/pages/profile-page/confirmation-modal/ConfirmationModal.jsx [hunk 21]
```diff
@@ -0,0 +1,37 @@
...
+import ConfirmationForm from './ConfirmationForm.jsx';
+import Dialog, {
+  DialogContent,
+  DialogTitle
+} from 'material-ui/Dialog';
+import 'assets/styles/Modal.css';
+
+import { getBEMClasses } from 'components/helpers/BEMHelper';
+
+const modal = 'modal';
```
Comment (k-stasevich) [inline client/src/components/pages/profile-page/confirmation-modal/ConfirmationModal.jsx]: stateless component

File: client/src/redux/selectors/filteredPosts.js [hunk 22]
```diff
@@ -0,0 +1,62 @@
+import { createSelector } from 'reselect';
+import { formValueSelector } from 'redux-form';
+
+const searchBarSelector = formValueSelector('searchBar');
+
+const postsSelector = state => state.posts.posts;
+const authUserSelector = state => state.user.user;
+const usersByIdSelector = state => state.profile.usersById;
+const reversedPostsSelector = posts => posts.reverse();
+const filteredByUserPostsSelector = (posts, user) => {
```
Comment (k-stasevich) [inline client/src/redux/selectors/filteredPosts.js]: UserPostsSelector

**Pull request merged**